### PR TITLE
add /v2/updates/price/latest endpoint

### PR DESCRIPTION
**Pull request by cctdaniel**
implement new v2 endpoint for /updates/price/latest which is a merge of the existing endpoints as follow:
- `/api/latest_price_feeds?ids[]=<price_feed_id>&ids[]=<price_feed_id_2>&..(&verbose=true)(&binary=true)`
- `/api/latest_vaas?ids[]=<price_feed_id>&ids[]=<price_feed_id_2>&...`

the proposed v2 endpoint is as follow:
- `/v2/updates/price/latest?ids[]=<price_feed_id>&ids[]=<price_feed_id_2>&..(&encoding=hex|base64)(&parsed=false)`

`encoding` defaults to `hex` and `parsed` defaults to `true` as per design guideline

this PR serves at an initial stab at the endpoint, will follow up with a separate PR for the tests

### CODE REVIEW

File: hermes/src/api/rest/latest_price_updates.rs [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+}
+
+/// Get the latest price updates by price feed id.
+///
+/// Given a collection of price feed ids, retrieve the latest Pyth price for each price feed.
+#[utoipa::path(
+    get,
+    path = "/v2/updates/price/latest",
+    responses(
+        (status = 200, description = "Price updates retrieved successfully", body = Vec<PriceUpdate>)
```
Comment (ali-bahjati) [inline hermes/src/api/rest/latest_price_updates.rs:66]: not found use case?

File: hermes/src/api/types.rs [hunk 2]
```diff
@@ -179,3 +189,81 @@ impl RpcPriceIdentifier {
...
+
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize, ToSchema)]
+pub struct PriceUpdate {
+    pub binary: BinaryPriceUpdate,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub parsed: Option<Vec<ParsedPriceUpdate>>,
+}
+
+impl PriceUpdate {
+    pub fn from_price_feed_update(
```
Comment (ali-bahjati) [inline hermes/src/api/types.rs:229]: Where is this used?

File: hermes/src/api/types.rs [hunk 3]
```diff
@@ -179,3 +189,81 @@ impl RpcPriceIdentifier {
...
+    Base64,
+    #[serde(rename = "hex")]
+    Hex,
+}
+
+impl Default for EncodingType {
+    fn default() -> Self {
+        EncodingType::Hex
+    }
+}
```
Comment (ali-bahjati) [inline hermes/src/api/types.rs:205]:
```suggestion
#[derive(Clone, Copy, Debug, Default, serde::Deserialize, serde::Serialize)]
pub enum EncodingType {
    #[serde(rename = "base64")]
    Base64,
    #[default]
    #[serde(rename = "hex")]
    Hex,
}
```

File: hermes/src/api/rest/latest_price_updates.rs [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+            e
+        );
+        RestError::UpdateDataNotFound
+    })?;
+
+    let price_updates: Vec<PriceUpdate> = price_feeds_with_update_data
+        .price_feeds
+        .into_iter()
+        .map(|price_feed| {
+            PriceUpdate::from_price_feed_update(price_feed, params.parsed, params.encoding)
```
Comment (ali-bahjati) [inline hermes/src/api/rest/latest_price_updates.rs:99]: this conversion doesn't make much sense to me. you probably need to implement `impl From<PriceFeedUpdate> for ParsedPriceUpdate` and you can just call it in the end when you know that the parsed data is needed.

File: hermes/src/api/rest/latest_price_updates.rs [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+        })
+        .collect();
+
+    let compressed_update_data = price_feeds_with_update_data.update_data;
```
Comment (ali-bahjati) [inline hermes/src/api/rest/latest_price_updates.rs:103]: i won't call it compressed, it's just the update data for the list of prices.

File: hermes/src/api/types.rs (hunk 2 above)
Comment (cctdaniel) [reply hermes/src/api/types.rs]: removed to use `ParsedPriceUpdate::from_price_feed_update`

File: hermes/src/api/types.rs [hunk 6]
```diff
@@ -58,6 +58,16 @@ pub struct RpcPriceFeedMetadata {
     pub prev_publish_time:          Option<UnixTimestamp>,
 }
 
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize, ToSchema)]
+pub struct RpcPriceFeedMetadataV2 {
+    #[schema(value_type = Option<u64>, example=85480034)]
+    pub slot:                 Option<Slot>,
+    #[schema(value_type = Option<i64>, example=doc_examples::timestamp_example)]
+    pub proof_available_time: Option<UnixTimestamp>,
```
Comment (ali-bahjati) [inline hermes/src/api/types.rs:66]: for the future: can we make this a ms-resolution timestamp?

**Review by ali-bahjati — APPROVED** [review]
LGTM. I left a comment on changing the `proof_available_timestamp` to have millisecond resolution which can be done in subsequent PRs.

**Pull request merged**